src/Router/BaseRouter.php

```diff
@@ -0,0 +1,99 @@
+<?php
+namespace LORIS\Router;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use \Psr\Http\Server\RequestHandlerInterface;
+use LORIS\Http\StringStream;
+
+// Handles the root of a LORIS install. It will mostly delegate to the
+// module router.
+// FIXME: Add other things in .htaccess here.
+class BaseRouter extends Prefix implements RequestHandlerInterface {
+    protected $projectdir;
+    protected $moduledir;
+    protected $user;
+
+    public function __construct(\User $user, string $projectdir, string $moduledir) {
+        $this->user = $user;
+        $this->projectdir = $projectdir;
+        $this->moduledir = $moduledir;
+    }
+
+    public function handle(ServerRequestInterface $request) : ResponseInterface {
+        $uri = $request->getUri();
+        $path = $uri->getPath();
+        // Replace multiple slashes in the URL with a single slash
+        $path = preg_replace("/\/+/", "/", $path);
+        // Remove a training slash remaining, so that foo/ and foo are the same
+        // route
+        $path = preg_replace("/\/$/", "", $path);
+        $request = $request->withAttribute("user", $this->user);
+        if ($path == "/" || $path == "") {
+            if ($this->user instanceof \LORIS\AnonymousUser) {
+                $modulename = "login";
+            }  else {
+                $modulename = "dashboard";
+            }
+            $request = $request->withURI($uri->withPath("/"));
+        } else if ($path[0] === "/") {
+            $path = substr($path, 1);
+            $request = $request->withURI($uri->withPath($path));
+        }
+
+        if (empty($modulename)) {
+            $components = preg_split("/\/+?/", $path);
```

Comment:
line #26 is taking cate of the multiple  `/`

Reply:
but it still needs to split? Is there any harm in using /+ instead of /{1} to split on?

Reply:
No.
Anyway, the `?` would need to be escaped if the `+` was removed.